oktapam/resource_security_policy.go

```diff
@@ -83,9 +87,48 @@ func resourceSecurityPolicy() *schema.Resource {
 							Description: descriptions.SecurityPolicyResources,
 							Elem: &schema.Resource{
 								Schema: map[string]*schema.Schema{
+									attributes.Secrets: {
+										Type:        schema.TypeList,
+										Optional:    true,
+										MinItems:    1,
+										MaxItems:    1,
+										Description: descriptions.SecurityPolicySecrets,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												attributes.Secret: {
```

Comment:
I think only Secret or SecretFolder can be specified but not both. You can use ConflictsWith property in the schema. Take a look at - https://github.com/okta/terraform-provider-oktapam/blob/8505915c86f1eb7fb65cfcf19b826fed65f55516/oktapam/resource_ad_certificate_request.go#L72

Reply:
Because we use TypeLists/TypeSets, using ConflictsWith is kind of a nightmare.  I ended up doing the validation within the code itself.